SaneEconomyCore/src/main/java/org/appledash/saneeconomy/SaneEconomy.java

```diff
@@ -44,50 +44,55 @@
         put("balancetop", new BalanceTopCommand(SaneEconomy.this));
     }};
 
-    public SaneEconomy() {
-        instance = this;
+    private final SanePlugin plugin;
+    private final boolean isBukkit;
+    private static SaneEconomy instance;
+
+    public SaneEconomy(SanePlugin plugin) {
+        isBukkit = Bukkit.getName().equals("Bukkit") || Bukkit.getName().equals("CraftBukkit");
+        this.plugin = plugin;
+        this.instance = this;
     }
 
-    @Override
-    public void onEnable() {
-        super.onEnable();
+    public void onLoad() {}
 
+    public void onEnable() {
         if (!loadConfig()) { /* Invalid backend type or connection error of some sort */
             shutdown();
             return;
         }
 
-        if (this.getConfig().getBoolean("locale-override", false)) {
+        if (plugin.getConfig().getBoolean("locale-override", false)) {
             Locale.setDefault(Locale.ENGLISH);
         }
 
         loadCommands();
         loadListeners();
 
-        if (getServer().getPluginManager().isPluginEnabled("Vault")) {
+        if (Bukkit.getServer().getPluginManager().isPluginEnabled("Vault")) {
             vaultHook = new VaultHook(this);
             vaultHook.hook();
-            getLogger().info("Hooked into Vault.");
+            log(Level.INFO, "Hooked into Vault.");
         } else {
-            getLogger().info("Not hooking into Vault because it isn't loaded.");
+            log(Level.INFO, "Not hooking into Vault because it isn't loaded.");
         }
 
-        if (this.getConfig().getBoolean("update-check", true)) {
-            versionChecker = new GithubVersionChecker("SaneEconomyCore", this.getDescription().getVersion().replace("-SNAPSHOT", ""));
-            this.getServer().getScheduler().runTaskAsynchronously(this, versionChecker::checkUpdateAvailable);
+        if (plugin.getConfig().getBoolean("update-check", true)) {
+            versionChecker = new GithubVersionChecker("SaneEconomyCore", plugin.getDescription().getVersion().replace("-SNAPSHOT", ""));
+            Bukkit.getServer().getScheduler().runTaskAsynchronously(plugin, versionChecker::checkUpdateAvailable);
```

Comment:
Can we use `plugin.getServer()` instead? I don't like the `Bukkit` static access methods for some reason.

Reply:
Yeah, that's fine- whatever works best for you. Originally I planned on using the plugin instead of the Bukkit static since most of your code was using it, but then I found a couple places where the Bukkit static methods were used so I just chose a method.